src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,289 @@
+            }
+            return returnValue;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Returns the closest barbarian to an entity
+     *
+     * @param entity             The entity to test against
+     * @param distanceFromEntity The distance to check for
+     * @return the barbarian (if any) that is nearest
+     */
+    public static Entity getClosestBarbarianToEntity(final Entity entity, final double distanceFromEntity)
+    {
+        final List<Entity> entityList = CompatibilityUtils.getWorld(entity).getEntitiesInAABBexcluding(
+          entity,
+          entity.getEntityBoundingBox().expand(
+            distanceFromEntity,
+            3.0D,
+            distanceFromEntity),
+          Entity::isEntityAlive);
+
+        final Optional<Entity> entityBarbarian = entityList.stream()
+                                                   .filter(BarbarianUtils::isBarbarian)
+                                                   .findFirst();
+
+        return entityBarbarian.orElse(null);
+    }
+
+    /**
+     * Simple method that returns whether or not an entity is a barbarian
+     *
+     * @param entity The entity to check
+     * @return Boolean value of whether the entity is a barbarian
+     */
+    public static Boolean isBarbarian(final Entity entity)
+    {
+        return (entity instanceof EntityBarbarian || entity instanceof EntityArcherBarbarian || entity instanceof EntityChiefBarbarian);
+    }
+
+    /**
+     * Returns whether a raid should happen depending on the Config
+     *
+     * @param world The world in which the raid is possibly happening (Used to get a random number easily)
+     * @return Boolean value on whether to act this night
+     */
+    public static boolean raidThisNight(final World world)
+    {
+        final float chance = (float) 1 / Configurations.averageNumberOfNightsBetweenRaids;
+        return world.rand.nextFloat() < chance;
+    }
+
+    /**
+     * Sets up and spawns the Barbarian entities of choice
+     *
+     * @param entityToSpawn  The entity which should be spawned
+     * @param numberOfSpawns The number of times the entity should be spawned
+     * @param spawnLocation  the location at which to spawn the entity
+     * @param world          the world in which the colony and entity are
+     */
+    private static void spawn(final ResourceLocation entityToSpawn, final int numberOfSpawns, final BlockPos spawnLocation, final World world)
+    {
+
+        if (spawnLocation != null && entityToSpawn != null && world != null)
+        {
+            if (!world.isBlockLoaded(spawnLocation))
+            {
+                ForgeChunkManager.Ticket chunkTicket = ForgeChunkManager.requestTicket(MineColonies.instance, world, ForgeChunkManager.Type.NORMAL);
+                if (chunkTicket == null)
+                {
+                    System.out.println("Well forge denied a Chunk Loading ticker, i guess you're not getting your horde tonight...");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS106)